t/24-worker-overall.t

```diff
@@ -103,6 +103,12 @@ combined_like { $worker->log_setup_info }
 qr/.*http:\/\/localhost:9527,https:\/\/remotehost.*qemu_i386,qemu_x86_64.*Errors occurred.*foo.*bar.*/s,
   'setup info with parse errors';
 
+if ($worker->settings->global_settings->{PACKAGES_CMD}) {
+    my $cmd = $worker->settings->global_settings->{PACKAGES_CMD};
+    ok(system($cmd. ' >/tmp/packages.txt') == 0,                'The PACKAGES_CMD runs');
+    ok(system('[ $(cat /tmp/packages.txt|wc -l) -gt 0 ]') == 0, 'The PACKAGES_CMD returns at least one package');
+}
+
```

Comment:
Sorry, but I don't get the point of this test.

* The test would not fail if the settings parser fails to set `$worker->settings->global_settings->{PACKAGES_CMD}`. It would make more sense to *assert* that `$worker->settings->global_settings->{PACKAGES_CMD}` is not empty.
* Executing the commands within the test *like* the worker would do does not test whether the worker would execute the commands as expected. You need to put the worker's code itself under test. Call the worker's init function (which you extended) instead of duplicating that code within the tests.
* We should not test `rpm` here. There's no need to do so and it makes the test suite less portable because rpm is a quite distribution-specific tool.
* Preferably use `is` to instead of `ok` to check whether to values are equal.

Reply:
The point is to test the command could be executed, but yes I agree that in a testing step could be not a good moment to do that because depends on the worker.ini configuration in the customer env. 
In my opinion, I should remove entirely this test and maybe the only could be tested here is that the global settings has the package cmd

Reply:
You could try to call the init function or you move the code into its own function and just call that (from init and from the test).